Makefile

```diff
@@ -8,10 +8,20 @@ COMPARE ?= 1
 NON_MATCHING ?= 0
 # If ORIG_COMPILER is 1, compile with QEMU_IRIX and the original compiler
 ORIG_COMPILER ?= 0
+# If COMPILER is GCC, compile with GCC instead of IDO.
+COMPILER ?= ido
+
+# If gcc is used, define the NON_MATCHING flag respectively so the files that
+# are safe to be used can avoid using GLOBAL_ASM which doesn't work with gcc.
+ifeq ($(COMPILER),gcc)
+  NON_MATCHING := 1
+  CPPFLAGS := -DCOMPILER_GCC
+  ZAPDFLAGS := --gcc-compat
```

Comment:
I think ideally there should be `ZAPDFLAGS ?=` at the top level (not in an if) to "declare" the variable (for makefile documentation purpose mostly), and this would be +=